src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/service/haflow/HaFlowUpdateService.java

```diff
@@ -162,9 +171,28 @@ public void handlePartialUpdateRequest(
             }
             carrier.sendNorthboundResponse(buildResponseMessage(
                     updateHaFlowResult.partiallyUpdatedHaFlow, commandContext));
+            FlowHistoryService.using(carrier).save(HaFlowHistory
```

Comment:
@niksv @yuliiamir 
this is an alternative way of how to save history on patch. Here we just wait until the decision of making the full update is made and the data is written to DB and only then save the "dump before". With this approach we don't save an extra dump before in case of the full update (the update makes its own dump before).
In contrast, in the partial update for the simple flow, we make a dump before _actually_ before any modifications are done. The problem there is that the decision and modifying the object is done after the dump is made, so we cannot not save the dump in case of the full update. It leads to the situation, when we save dump before two times. However, they are different dumps, because we can first modify some fields in the partial update and then pass the modified flow to the full update.

What do you think? Should make one way or another in both places? Should we fix it somehow differently in both cases? (move the decision to earlier stage, before modifying the flow)?

Reply:
I think 2 "before" is better, because we will have more info for an operator